torchgeo/datasets/usavars.py

```diff
@@ -0,0 +1,241 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""USAVars dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Optional, Sequence
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import download_url, extract_archive
+
+
+class USAVars(VisionDataset):
+    """USAVars dataset.
+
+    The USAVars dataset is reproduction of the dataset used in the paper "`A
+    generalizable and accessible approach to machine learning with global satellite
+    imagery <https://doi.org/10.1038/s41467-021-24638-z>`_". Specifically, this dataset
+    includes 1 sq km. crops of NAIP imagery resampled to 4m/px cenetered on ~100k points
+    that are sampled randomly from the contiguous states in the USA. Each point contains
+    three continous valued labels (taken from the dataset released in the paper): tree
+    cover percentage, elevation, and population density.
+
+    Dataset format:
+    * images are 4-channel GeoTIFFs
+    * labels are singular float values
+
+    Dataset labels:
+    * tree cover
+    * elevation
+    * population density
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1038/s41467-021-24638-z
+
+    .. versionadded:: 0.3
+    """
+
+    url_prefix = (
+        "https://files.codeocean.com/files/verified/"
+        + "fa908bbc-11f9-4421-8bd3-72a4bf00427f_v2.0/data/int/applications"
+    )
+    pop_csv_suffix = "CONTUS_16_640_POP_100000_0.csv?download"
+    uar_csv_suffix = "CONTUS_16_640_UAR_100000_0.csv?download"
+
+    data_url = "https://mosaiks.blob.core.windows.net/datasets/uar.zip"
+    dirname = "uar"
+
+    md5 = "677e89fd20e5dd0fe4d29b61827c2456"
+
+    label_urls = {
+        "housing": f"{url_prefix}/housing/outcomes_sampled_housing_{pop_csv_suffix}",
+        "income": f"{url_prefix}/income/outcomes_sampled_income_{pop_csv_suffix}",
+        "roads": f"{url_prefix}/roads/outcomes_sampled_roads_{pop_csv_suffix}",
+        "nightligths": f"{url_prefix}/nightlights/"
+        + f"outcomes_sampled_nightlights_{pop_csv_suffix}",
+        "population": f"{url_prefix}/population/"
+        + f"outcomes_sampled_population_{uar_csv_suffix}",
+        "elevation": f"{url_prefix}/elevation/"
+        + f"outcomes_sampled_elevation_{uar_csv_suffix}",
+        "treecover": f"{url_prefix}/treecover/"
+        + f"outcomes_sampled_treecover_{uar_csv_suffix}",
+    }
+
+    # ALL_LABELS = label_urls.keys()
+    ALL_LABELS = ["treecover", "elevation", "population"]
```

Comment:
Which of these is right? Should we delete the commented out line? Can we add an assert to ensure that the `labels` a user asks for is valid?